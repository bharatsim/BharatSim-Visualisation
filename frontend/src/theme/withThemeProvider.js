import React from 'react';
import { MuiThemeProvider, CssBaseline } from '@material-ui/core';

import theme from './theme';

function withThemeProvider(WrappedComponent) {
  return function WithTheme(props) {
    return (
      <MuiThemeProvider theme={theme}>
        <CssBaseline />
        <WrappedComponent {...props} />
      </MuiThemeProvider>
    );
  };
}

export default withThemeProvider;
