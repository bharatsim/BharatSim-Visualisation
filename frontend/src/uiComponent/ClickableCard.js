import React from 'react';
import Box from '@material-ui/core/Box';
import { makeStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import { ChildrenPropTypes } from '../commanPropTypes';

const useStyles = makeStyles((theme) => {
  return {
    ClickableCardContainer: {
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center',
      alignItems: 'center',
      border: '1px dashed',
      borderRadius: theme.spacing(1),
      borderColor: theme.colors.primaryColorScale['200'],
      '&:hover': {
        backgroundColor: theme.colors.primaryColorScale['50'],
        cursor: 'pointer',
      },
    },
  };
});

function ClickableCard({ children, onClick }) {
  const classes = useStyles();
  return (
    <Box onClick={onClick} className={classes.ClickableCardContainer}>
      {children}
    </Box>
  );
}

ClickableCard.propTypes = {
  onClick: PropTypes.func.isRequired,
  children: ChildrenPropTypes.isRequired,
};

export default ClickableCard;
