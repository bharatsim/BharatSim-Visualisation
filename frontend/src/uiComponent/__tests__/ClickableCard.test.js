import React from 'react';
import { render } from '@testing-library/react';
import { fireEvent } from '@testing-library/dom';
import ClickableCard from '../ClickableCard';
import withThemeProvider from '../../theme/withThemeProvider';

describe('<ClickableCard />', () => {
  const ClickableCardComponent = withThemeProvider(ClickableCard);
  it('should match snapshot', () => {
    const { container } = render(
      <ClickableCardComponent onClick={jest.fn()}>Sample Test Component</ClickableCardComponent>,
    );
    expect(container).toMatchSnapshot();
  });
  it('should call onclick function on click of component', () => {
    const mockClickFunction = jest.fn();
    const { getByText } = render(
      <ClickableCardComponent onClick={mockClickFunction}>
        Sample Test Component
      </ClickableCardComponent>,
    );
    fireEvent.click(getByText('Sample Test Component'));

    expect(mockClickFunction).toHaveBeenCalled();
  });
});
