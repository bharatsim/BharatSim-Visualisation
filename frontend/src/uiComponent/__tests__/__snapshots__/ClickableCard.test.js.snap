// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClickableCard /> should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-2 makeStyles-ClickableCardContainer-1"
  >
    Sample Test Component
  </div>
</div>
`;
